dandi/__init__.py

```diff
@@ -36,3 +36,57 @@ def set_logger_level(lgr, level):
 set_logger_level(lgr, os.environ.get("DANDI_LOG_LEVEL", logging.INFO))
 FORMAT = "%(asctime)-15s [%(levelname)8s] %(message)s"
 logging.basicConfig(format=FORMAT)
+
+
+def check_latest_version(raise_exception=False):
+    """
+    Check for the latest version of the library.
+    Parameters
+    ----------
+    raise_exception: :obj:`bool`
+        Raise a RuntimeError if a bad version is being used
+    """
+    import etelemetry
```

Comment:
The whole function should get under try:except so we don't blow if eg etelemetry import blows or there is a bug in finally handling below

I could add a decorator later

Reply:
for this last one, since etelemetry is now a dependency i would not put it under try/except. we don't do it for any other import of package on which something is dependent.

Reply:
if something like `yaml` fails to import -- major functionality of the package is impaired.  If `etelemetry` fails to import -- who really cares?  package should remain fully functional (unless it is a very very bad version ;) ) that is the difference for me.
BTW -- debian folks even already packaged it but it is waiting for acceptance -- lets see if debian ftp masters would raise a red flag ;) https://ftp-master.debian.org/new/python-etelemetry_0.1.2-1.html

Reply:
since the intent of etelemetry is also to notify bad versions, it's important that it gets imported. it's not just a version check. so i think people should care.

Reply:
"should care" != "total lock down, can't use any functionality". and situation is not much different from a user working offline -- no checks could be done, so should he stop? no and the other exception handler allows for that.  So I do not see how the situation of "I failed to import etelemetry" is different.
IMHO etelemetry should be as non-intrusive as possible.

Reply:
if a package is in the requirements, then it should not be needed to be caught. if we say it is optional it's a separate thing. this PR started with this being a necessary thing for users. so it's in the requirements and therefore should not have a need to be try-excepted.

Reply:
we will chat about this over some beers (hopefully this year ;-))